### PR TITLE
[Doc] update partition desc in alter table

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 1]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+  <TabItem value="example1" label="手动创建分区" default>
+
+  - 仅指定各个分区的上界
+
+    语法：
+
+    ```sql
+    PARTITION BY RANGE ( <partitioning_column1> [, <partitioning_column2>, ... ] )
+    (
+    PARTITION <partition_name1> VALUES LESS THAN ("<upper_bound_for_partitioning_column1>" [ , "<upper_bound_for_partitioning_column2>", ... ] )
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:130]: 给个缩进？

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 2]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+      ```
 
-2. 分区为左闭右开区间，如果用户仅指定右边界，系统会自动确定左边界。
-3. 如果没有指定分桶方式，则自动使用建表使用的分桶方式。
-4. 如指定分桶方式，只能修改分桶数，不可修改分桶方式或分桶列。
-5. `["key" = "value"]` 部分可以设置分区的一些属性，具体说明见 [CREATE TABLE](./CREATE_TABLE.md)。
+    - 如果分区列为整数类型
+
+      ```sql
+      START ("<start_integer>") END ("<end_integer>") EVERY (<partitioning_granularity>)
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:165]: 建议加个注意说明：整形也需要用引号，以按照字符串的方式传入。

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 3]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+
+- List 分区
+
+    语法：
+
+    ```sql
+    PARTITION <partition_name> VALUES IN (value_list) [, (value_list) ...]
+
+    value_list ::=
+        value_item [, value_item [, ...] ]
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:186]: value_item [, value_item  ... ]

我们其他地方应该都是这种方式，很少用  `a [, a [, ...] ]` 这种方式。

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 4]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+    - 可以通过 `["key" = "value"]` 为新的分区设置属性，具体说明见 [CREATE TABLE](./CREATE_TABLE.md#properties)。
+
+    :::
+
+- List 分区
+
+    语法：
+
+    ```sql
+    PARTITION <partition_name> VALUES IN (value_list) [, (value_list) ...]
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:183]: 这处语法看起来不对啊。应该是：
`... VALUES IN (value_list)`
不需要后面的 `[, (value_list) ...]`

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 5]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+
+    value_list ::=
+        value_item [, value_item [, ...] ]
+
+
+    value_item ::=
+        { <value> | ( <value> [, <value>, [, ...] ] ) }
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:190]: { <value> | ( <value> [, <value> ...] ) }

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 6]
```diff
@@ -104,22 +107,109 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
 
 ```SQL
 ALTER TABLE [<db_name>.]<tbl_name> 
-ADD PARTITION [IF NOT EXISTS] <partition_name>
-partition_desc ["key"="value"]
-[DISTRIBUTED BY HASH (k1[,k2 ...]) [BUCKETS num]];
+ADD PARTITION [IF NOT EXISTS] <partition_name> partition_desc ["key"="value"]
+[distribution_desc];
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:111]: `distribution_desc` 要不要至少给个链接

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 7]
```diff
@@ -103,113 +101,97 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
+        PARTITION [IF NOT EXISTS] <partition_name> VALUES partitionKeyDesc
 
-  - 仅指定各个分区的上界
+    partitionKeyDesc ::=
+        { LESS THAN (MAXVALUE | partitionValueList)
+        | [ partitionValueList , partitionValueList ) } -- 注意此处的 [ 代表左闭合区间
 
-    语法：
+    partitionValueList ::=
+        ( partitionValue [, ...] )
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:122]: partitionValue 是什么？

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 8]
```diff
@@ -103,113 +101,97 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
 
+语法：
+
 - Range 分区
 
-  <Tabs groupId="manual partitioning">
-  <TabItem value="example1" label="手动创建分区" default>
+    ```SQL
+    ALTER TABLE
+        ADD { single_rangePartition | PARTITIONS multiRangePartition } [distributionDesc] ["key"="value"];
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:112]: 这个 PARTITIONS 直接放到 multiRangePartition 才看起来协调些

File: docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 9]
```diff
@@ -103,113 +101,97 @@ ALTER TABLE [<db_name>.]<tbl_name> COMMENT = "<new table comment>";
...
-    PARTITION BY RANGE ( <partitioning_column1> [, <partitioning_column2>, ... ] )
-    (
-    PARTITION <partition_name1> VALUES [ ( "<lower_bound_for_partitioning_column1>" [ , "<lower_bound_for_partitioning_column2>", ... ] ), ( "<upper_bound_for_partitioning_column1>" [ , "<upper_bound_for_partitioning_column2>", ... ] ) ) 
-    [ ,
-    PARTITION <partition_name2> VALUES [ ( "<lower_bound_for_partitioning_column1>" [ , "<lower_bound_for_partitioning_column2>", ... ] ), ( "<upper_bound_for_partitioning_column1>" [ , "<upper_bound_for_partitioning_column2>", ... ] ) ) 
-    , ...]
-    )
-    ```
+    value_list ::=
+        value_item [, ...] -- 如果存在多个枚举值，则枚举值之间用英文逗号分隔。
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:136]: 这里不需要这个解释，多余。（`枚举值之间用英文逗号分隔。` SQL语法中不可能用中文符号的）

Comment (jaogoy) [reply docs/zh/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:136]: 下同

**Review by jaogoy — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by jaogoy** [comment]
ignore backport check: 2.5.22

**Pull request merged**